src/libs/API.js

```diff
@@ -18,7 +18,40 @@ import CONST from '../CONST';
  * @param {Object} [onyxData.successData] - Onyx instructions that will be passed to Onyx.update() when the response has jsonCode === 200.
  * @param {Object} [onyxData.failureData] - Onyx instructions that will be passed to Onyx.update() when the response has jsonCode !== 200.
  */
-function write(command, apiCommandParameters = {}, onyxData = {}) {
+function write(command, apiCommandParameters = {}, onyxData = {}, usePromise = false) {
```

Comment:
I don't think this is something we would want to add, I thought we decided on not using promises with this for a specific reason. cc @tgolen @yuwenmemon @marcaaron

Reply:
Without looking into the details of what we are trying to accomplish with this PR...

This is pretty much the whole reason why we have `makeRequestWithSideEffects()`. 

`API.write()` should not return a promise and in general we should avoid chaining things.

Based on the fact that this PR is in a state open for review - I think we maybe missed something somewhere and should most likely move this conversation to Slack to get wider input before changing anything about our architecture.

Reply:
Sorry @marcaaron This is definitely just a draft PR at this point. I'm open to better solutions to fix this [issue](https://github.com/Expensify/App/issues/12714).

Reply:
Yeah I don't see this as a viable solution either, and `makeRequestWithSideEffects` seems to be only used when we deal with network requests, unless I'm missing something? So that leaves us with either the debounce solution or figuring out something else. I agree we should bring this discussion to Slack.